src/Umbraco.Cms.Api.Delivery/Services/ApiContentQueryService.cs

```diff
@@ -85,20 +92,31 @@ public PagedModel<Guid> ExecuteQuery(string? fetch, IEnumerable<string> filters,
                 return null;
             }
 
-            var value = string.IsNullOrWhiteSpace(selector.Value) == false
+            fieldName = selector.FieldName;
+            fieldValue = string.IsNullOrWhiteSpace(selector.Value) == false
                 ? selector.Value
                 : _fallbackGuidValue;
-            queryOperation = baseQuery.Field(selector.FieldName, value);
         }
         else
         {
-            // TODO: If no params or no fetch value, get everything from the index - make a default selector and register it by the end of the collection
-            // TODO: This selects everything without regard to the current start-item header - make sure we honour that if it is present
-            // This is a temp Examine solution
-            queryOperation = baseQuery.Field("__IndexType", "content");
+            // If no params or no fetch value, get everything from the index - this is a way to do that with Examine
```

Comment:
To avoid nesting conditional logic here, you can make the `fieldName` and `fieldValue` fields nullable:

```cs
string? fieldName = null;
string? fieldValue = null;
```

This allows you to replace the `else` statement with something like this:

```cs
if (fieldName == null && _requestStartItemProviderAccessor.TryGetValue(out IRequestStartItemProvider? requestStartItemProvider))
{
    IPublishedContent? startItem = requestStartItemProvider.GetStartItem();
    if (startItem is not null)
    {
        // Reusing the boolean operation of the "Descendants" selector, as we want to get all the nodes from the given starting point
        fieldName = DescendantsSelectorIndexer.FieldName;
        fieldValue = startItem.Key.ToString();
    }
}
```

...and then you can perform the fallback check at the end of this method, just before returning the field query. Something like:

```cs
// If no params or no fetch value, get everything from the index - this is a way to do that with Examine
fieldName ??= UmbracoExamineFieldNames.CategoryFieldName;
fieldValue ??= "content";

return baseQuery.Field(fieldName, fieldValue);
```

Reply:
This results in a quite clean implementation 🙌 thanks for the suggestion